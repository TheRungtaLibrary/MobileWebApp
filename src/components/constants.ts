export const size200 = 200;

export const size640 = 640;