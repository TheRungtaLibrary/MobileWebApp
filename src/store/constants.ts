export const Action_Types = {
    GET_VEHICLE_DATA_REQUEST: 'GET_VEHICLE_DATA_REQUEST',
    GET_VEHICLE_DATA_SUCCESS: 'GET_VEHICLE_DATA_SUCCESS'
};