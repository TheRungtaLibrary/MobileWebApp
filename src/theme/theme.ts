export const theme = {
    palette: {
        primary: {
            main: '#0052cc',
        },
        secondary: {
            main: '#edf2ff',
        },
        tertiary: {
            main: '#F8F8F8',
            light: '#ffffff',
        },
        background: {
            main: '#f6ded2',
            light: '#000',
        },
        border: {
            main: '#efefef',
            light: '#808080',
        },
        text: {
            main: '#ebedf2',
        }
    },
};