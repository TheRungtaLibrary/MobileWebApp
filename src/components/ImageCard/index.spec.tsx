import React, { FunctionComponent } from 'react';
import { render, screen } from '@testing-library/react';
import { ImageCard, Props as ImageCardProps } from './index';
import { size200 } from '../constants';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../theme/theme';

describe('Given <ImageCard />', () => {
    let defaultProps: ImageCardProps;
    let component: HTMLElement;
    let Page: FunctionComponent<any>;

    beforeAll(() => {
        defaultProps = {
            uri: '',
            idx: 1,
            onClickHandler: () => { },
        };
    })

    beforeEach(() => {
        Page = (): JSX.Element => (
            <ThemeProvider theme={theme}>
                <ImageCard {...defaultProps} />
            </ThemeProvider>
        );

        const { container } = render(<Page />);

        component = container;
    })

    describe('When it initializes', () => {
        it('matches snapshot', () => {
            expect(component).toMatchSnapshot();
        });
    });

    it('renders image of thumbnail size', () => {
        const image = screen.getByRole('img') as HTMLImageElement;
        expect(image.src).toContain(size200.toString());
    });
});
