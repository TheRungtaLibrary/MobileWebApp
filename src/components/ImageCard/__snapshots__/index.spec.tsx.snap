// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <ImageCard /> When it initializes matches snapshot 1`] = `
<div>
  <div
    class="sc-bdvvtL bdgXaM"
    data-testid="image-card"
  >
    <figure>
      <img
        alt="image"
        src="https://?rule=mo-200.jpg"
      />
      <figcaption
        class="sc-gsDKAQ dCXWnf"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
        >
          photo 
          2
        </h3>
      </figcaption>
    </figure>
  </div>
</div>
`;
